types.go

```diff
@@ -131,3 +132,50 @@ type PrivateSectorInfo struct {
 type AllocationManager interface {
 	Free()
 }
+
+func DeDuplicatePrivateSectorInfo(ctx context.Context, sortPrivSectors *SortedPrivateSectorInfo) (SortedPrivateSectorInfo, error) {
```

Comment:
Looking at code in Lotus it looks like it should be possible to execute this and split operations before creating the `SortedPrivateSectorInfo`.
I strongly suggest changing it to that.
The SortedPrivateSectorInfo type should only be used for directly passing sector information into FFI not for carrying sector information around or operating on them.